src/MongoDB.Driver.Core/Core/Operations/ChangeStreamCursor.cs

```diff
@@ -155,7 +154,17 @@ private IEnumerable<TDocument> DeserializeDocuments(IEnumerable<RawBsonDocument>
 
             if (lastRawDocument != null)
             {
-                _changeStreamOperation.ResumeAfter = lastRawDocument["_id"].DeepClone().AsBsonDocument;
+                var resumeToken = lastRawDocument["_id"].DeepClone().AsBsonDocument;
+
+                if (_changeStreamOperation.StartAfter != null && _changeStreamOperation.ResumeAfter == null)
+                {
+                    _changeStreamOperation.StartAfter = resumeToken;
+                }
+                else
+                {
+                    _changeStreamOperation.ResumeAfter = resumeToken;
+                }
+
```

Comment:
As I mentioned in an earlier code review I would like us to NOT alter any fields originally set by the creator of the operation. Instead, just store the resume token in a new field:

    if (lastRawDocument != null)
    {
        _changeStreamOperation.DocumentResumeToken = lastRawDocument["_id"].DeepClone().AsBsonDocument;
    }

Reply:
Done